ERIC.js

```diff
@@ -30,77 +30,102 @@
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-function detectWeb(doc,url) {
-	var xpath='//meta[@name="citation_journal_title"]';
-	var type = ZU.xpathText(doc, '//meta[@name="source"]/@content');	
-	if (ZU.xpath(doc, xpath).length > 0) {
-		if (type && type.indexOf("Non-Journal")!=-1) return "book"
-		else return "journalArticle";
+
+function detectWeb(doc, url) {
+	var hasTitle = doc.querySelector("meta[name=citation_title]");
+	var type = doc.querySelector("meta[name=source][content]");
```

Comment:
I know you're just porting, but this should go within the`if (hasTitle)` block so that it doesn't run unnecessarily.